propagation_test.go

```diff
@@ -0,0 +1,57 @@
+package opentracing
+
+import (
+	"net/http"
+	"testing"
+)
+
+const testHeaderPrefix = "testprefix-"
+
+func TestHTTPHeaderInject(t *testing.T) {
```

Comment:
this test doesn't feel right, its logic relies mostly on the details of the test tracer implementation.  It looks more like a testable example (https://blog.golang.org/examples). For the actual code in the API, I think we should simply test using the HTTP carrier to put stuff and read stuff back, without involving the tracer.